include/boost/math/tools/polynomial.hpp

```diff
@@ -558,59 +562,52 @@ class polynomial :
 
 
 template <class T>
-inline polynomial<T> operator + (const polynomial<T>& a, const polynomial<T>& b)
+inline polynomial<T> operator + (polynomial<T> a, const polynomial<T>& b)
 {
-   polynomial<T> result(a);
-   result += b;
-   return result;
+   a += b;
+   return a;
```

Comment:
I'm not completely certain about this, but I believe we need a:

return std::move(a);

Here to be sure that we don't make an accidental copy.  If the function is expanded inline it will make not one jot of difference of course, but IMO we can't rely on that.

To make it C++03 compatible we would need something like:

#ifndef BOOST_NO_RVALUE_REFERENCES
#  define BOOST_MATH_MOVE(x) std::move(x)
#else 
#  define BOOST_MATH_MOVE(x) x
#endif  

Then 

return BOOST_MATH_MOVE(a);